File: 008-increasing-triplet/Solution.java
class Solution {
    public boolean increasingTriplet(int[] nums) {
        if(nums == null || nums.length < 3) {
            return false;
        }

        int smallest = Integer.MAX_VALUE;
        int biggest = Integer.MAX_VALUE;

        for(var num: nums) {
            if(num <= smallest) {
                smallest = num;
            } else if (num <= biggest) {
                biggest = num;
            } else {
                return true;
            }
        }
        
        return false;
    }
}

File: 008-increasing-triplet/Solution.swift
class Solution {
    func increasingTriplet(_ nums: [Int]) -> Bool {
        if nums.count < 3 {
            return false
        }

        var smallest = Int.max
        var largest = Int.max

        for num in nums {
            if num <= smallest {
                smallest = num
            } else if (num <= largest) {
                largest = num
            } else {
                return true
            }
        }

        return false
    }
}